src/main/java/games/strategy/engine/framework/startup/ui/PlayerSelectorRow.java

```diff
@@ -73,26 +77,26 @@ public void actionPerformed(final ActionEvent e) {
     }
     if (!(previousSelection.equals("no_one")) && Arrays.asList(types).contains(previousSelection)) {
       playerTypes.setSelectedItem(previousSelection);
-    } else if (player.isAiDefault()) {
+    } else if (player.getDefaultType().equals(PLAYER_TYPE_AI)) {
```

Comment:
You might want to swap the order of the comparison here because `PlayerID#getDefaultType()` can return `null`.  It looks like that should never happen for `PlayerID` instances created via `GameParser`, but there is a public constructor which does set the corresponding field to `null` (although it appears it's only used by the tests at this time).